File: 6. BitwiseAdder/bitwiseadder.py
def bitwise_add(a, b):
    sum = a ^ b
    carry = a & b
    while(carry != 0):
        carry <<= 1
        carryHolder = carry
        carry = sum & carry
        sum = sum ^ carryHolder
    return sum

def main():
    print(bitwise_add(4,10));

if __name__ == "__main__":
    main()

File: 7_BitwiseAdder/bitwiseadder.c
#include <stdio.h>

int add(int a, int b){
  int sum = a ^ b;
  int carry = a & b;
  while(carry != 0){
    carry = carry << 1;
    int temp = sum ^ carry;
    carry = sum & carry;
    sum = temp;
  }
  return sum;
}

int main(){
  int a, b;
  printf("Enter a number: ");
  scanf("%d", &a);
  printf("Enter another number: ");
  scanf("%d", &b);
  printf("%d", add(a,b));
}
